node/node_test.go

```diff
@@ -1326,9 +1344,31 @@ func TestNode_GetESDTsWithRole(t *testing.T) {
 	require.Len(t, tokenResult, 0)
 }
 
-func TestNode_GetESDTsRoles(t *testing.T) {
+func TestNode_GetESDTsWithRole_ShouldWork(t *testing.T) {
+	t.Parallel()
+
+	testNodeGetESDTsWithRole(t, node.NewNodeFactory(), core.MetachainShardId)
+}
+
+func TestNode_GetESDTsWithRole_WrongChainId(t *testing.T) {
 	t.Parallel()
 
+	coreComponents := getDefaultCoreComponents()
+	processComponents := getDefaultProcessComponents()
+	processComponents.ShardCoord = &mock.ShardCoordinatorMock{
+		SelfShardId: 0,
+	}
+	n, err := node.NewNode(
+		node.WithCoreComponents(coreComponents),
+		node.WithProcessComponents(processComponents))
+	require.Nil(t, err)
+
+	testResult, _, err := n.GetESDTsWithRole(testscommon.TestAddressAlice, core.ESDTRoleNFTCreate, api.AccountQueryOptions{}, context.Background())
```

Comment:
If found suitable for you as well, we could have all of these vars called like this
```suggestion
	res, _, err := n.GetESDTsWithRole(testscommon.TestAddressAlice, core.ESDTRoleNFTCreate, api.AccountQueryOptions{}, context.Background())
```